Source/WebCore/page/Navigation.cpp

```diff
@@ -169,11 +169,11 @@ void Navigation::initializeForNewWindow(std::optional<NavigationNavigationType>
                     m_entries[*previousNavigation->m_currentEntryIndex] = NavigationHistoryEntry::create(*this, *currentItem);
 
                 m_currentEntryIndex = getEntryIndexOfHistoryItem(m_entries, *currentItem);
-
-                ASSERT(navigationType);
-                m_activation = NavigationActivation::create(*navigationType, *currentEntry(), WTFMove(previousEntry));
-
-                return;
+                if (m_currentEntryIndex) {
+                    RELEASE_ASSERT(navigationType);
```

Comment:
Please drop this assertion. There is already such assertion in std::optional's `operator*` since we're using libcxx in hardened mode.